### ISSUE TITLE
feat: bind Interval data type to Python timedelta

### ISSUE THREAD

**Issue by rad-pat** [databendlabs/bendsql#552]
I'm trying to introduce Interval datatype to SQLAlchemy dialect which uses BendSQL as a driver. 
Can we please have the Interval type map to a Python `datetime.timedelta`, currently it is returned as string and causes test failures on the SQLAlchemy side, e.g.
`AssertionError: ('24:00:04',) != (datetime.timedelta(days=1, seconds=4),)`

**Comment by everpcpc** [comment]
cc @TCeason could you help with this?

**Comment by rad-pat** [comment]
FYI, I have worked around this for the time being by parsing the string result: https://github.com/PlaidCloud/databend-sqlalchemy/blob/interval/databend_sqlalchemy/types.py#L88..L109

**Comment by TCeason** [comment]
> FYI, I have worked around this for the time being by parsing the string result: https://github.com/PlaidCloud/databend-sqlalchemy/blob/interval/databend_sqlalchemy/types.py#L88..L109

Yes. The string format is single. We can add same parse logic in bendsql.
